src/diffusers/loaders/lora_pipeline.py

```diff
@@ -294,22 +294,15 @@ def load_lora_into_unet(
                 "`low_cpu_mem_usage=True` is not compatible with this `peft` version. Please update it with `pip install -U peft`."
             )
 
-        # If the serialization format is new (introduced in https://github.com/huggingface/diffusers/pull/2918),
-        # then the `state_dict` keys should have `cls.unet_name` and/or `cls.text_encoder_name` as
-        # their prefixes.
-        keys = list(state_dict.keys())
-        only_text_encoder = all(key.startswith(cls.text_encoder_name) for key in keys)
-        if not only_text_encoder:
-            # Load the layers corresponding to UNet.
-            logger.info(f"Loading {cls.unet_name}.")
-            unet.load_lora_adapter(
-                state_dict,
-                prefix=cls.unet_name,
-                network_alphas=network_alphas,
-                adapter_name=adapter_name,
-                _pipeline=_pipeline,
-                low_cpu_mem_usage=low_cpu_mem_usage,
-            )
```

Comment:
We're handling all of this within the `load_lora_adapter()` method now which I think is more appropriate as:

1. It takes care of logging when users try to load LoRA into a model via `pipe.load_lora_weights()`. 
2. Users try to load LoRAs directly into a model with the `load_lora_adapter()` method (with something like `unet.load_lora_adapter()`. 

Helps to avoid duplication. I have run the integration tests, too and nothing is breaking due to this.